### PR TITLE
Fix a bug in Adaptive_Three_Shift

### PR DESCRIPTION
**Pull request by chrislzhao**
@ye-luo This PR fixes a bug that affects the correctness of correlated sampling in the adaptive_three_shift method.

### CODE REVIEW

**Comment by markdewing** [comment]
okay to test

File: src/QMCDrivers/QMCFixedSampleLinearOptimize.cpp [hunk 1]
```diff
@@ -1051,7 +1051,7 @@ bool QMCFixedSampleLinearOptimize::adaptive_three_shift_run()
     for (int j = 0; j < parameterDirections.size(); j++)
     {
       if (j != central_index)
-        parameterDirections.at(j).at(i + 1) -= parameterDirections.at(1).at(i + 1);
```
Comment (ye-luo) [inline src/QMCDrivers/QMCFixedSampleLinearOptimize.cpp]: Please remove the following commented line.

Comment (chrislzhao) [reply src/QMCDrivers/QMCFixedSampleLinearOptimize.cpp]: Done

**Comment by ye-luo** [comment]
Please explain what is the bug and how the correct is affected. Everything else looks good.

**Comment by chrislzhao** [comment]
@ye-luo In the adaptive three shift method, three different shift values are used to stabilize the LM eigensolver, and leads to three different sets of wave function parameter updates. It will then do a correlated sampling with wave function parameters updated based on the medium-sized shift as the guiding function, in order to decide which shift is the best (gives the lowest cost function). This bug affects the correlated sampling part, in which the code will need to loop over shifts and update the wave function to make sure the wave functions are updated correctly based on that shift. Within the loop, it needs to check whether the shift is the medium sized one, and if it is then there is no needs to update the wave function. However, this check was wrongly coded. Instead of check "k == central_index", it does "k == num_shifts". Therefore the code used an incorrect wave function as the medium shift updated wave function. 

Another change I made would not change the current code. At line 1054 I simply change parameterDirections.at(1) to parameterDirections.at(central_index). Since 3 shifts are used, central_index is 1. But this may be a problem in the future when we decide to use more shifts.

**Pull request merged**